src/main.rs

```diff
@@ -930,53 +963,91 @@ pub fn non_zero_mixed_small_non_canonical() -> Result<(TestVector, TestVector)>
         pub_key: pub_key_arr,
         signature: serialize_signature(&r, &s),
     };
+    vec.push(tv2);
 
-    Ok((tv1, tv2))
+    Ok(vec)
 }
 
 fn generate_test_vectors() -> Result<Vec<TestVector>> {
-    // #0-1: canonical S, small R, small A
+    let mut info = Builder::default();
+    info.append("|  |    msg |    sig |  S   |    A  |    R  | cof-ed | cof-less |        comment        |\n");
+    info.append("|---------------------------------------------------------------------------------------|\n");
     let mut vec = Vec::new();
-    let (tv1, tv2) = zero_small_small().unwrap();
-    vec.push(tv1); // passes cofactored, fails cofactorless
+
+    // #0: canonical S, small R, small A
+    let (_tv1, tv2) = zero_small_small().unwrap();
+    info.append(format!(
+        "| 0| ..{:} | ..{:} |  = 0 | small | small |    V   |    V     | small A and R |\n",
+        &hex::encode(&tv2.message)[60..],
+        &hex::encode(&tv2.signature)[124..]
+    ));
     vec.push(tv2); // passes cofactored, passes cofactorless
 
-    // #2-3: canonical S, mixed R, small A
-    let (tv1, tv2) = non_zero_mixed_small().unwrap();
-    vec.push(tv1); // passes cofactored, fails cofactorless
+    // #1: canonical S, mixed R, small A
+    let (_tv1, tv2) = non_zero_mixed_small().unwrap();
+    info.append(format!(
+        "| 1| ..{:} | ..{:} |  < L | small | mixed |    V   |    V     | small A only |\n",
+        &hex::encode(&tv2.message)[60..],
+        &hex::encode(&tv2.signature)[124..]
+    ));
     vec.push(tv2); // passes cofactored, passes cofactorless
 
-    // #4-5: canonical S, small R, mixed A
-    let (tv1, tv2) = non_zero_small_mixed().unwrap();
-    vec.push(tv1); // passes cofactored, fails cofactorless
+    // #2: canonical S, small R, mixed A
+    let (_tv1, tv2) = non_zero_small_mixed().unwrap();
+    info.append(format!(
+        "| 2| ..{:} | ..{:} |  < L | mixed | small |    V   |    V     | small R only |\n",
+        &hex::encode(&tv2.message)[60..],
+        &hex::encode(&tv2.signature)[124..]
+    ));
     vec.push(tv2); // passes cofactored, passes cofactorless
 
-    // #6-7: canonical S, mixed R, mixed Axs
+    // #3-4: canonical S, mixed R, mixed A
     let (tv1, tv2) = non_zero_mixed_mixed().unwrap();
+    info.append(format!("| 3| ..{:} | ..{:} |  < L | mixed | mixed |    V   |    V     | succeeds unless full-order is checked |\n", &hex::encode(&tv2.message)[60..], &hex::encode(&tv2.signature)[124..]));
     vec.push(tv2); // passes cofactored, passes cofactorless
+    info.append(format!(
+        "| 4| ..{:} | ..{:} |  < L | mixed | mixed |    V   |    X     |  |\n",
+        &hex::encode(&tv1.message)[60..],
+        &hex::encode(&tv1.signature)[124..]
+    ));
     vec.push(tv1); // passes cofactored, fails cofactorless
```

Comment:
The order of these two pushes ((tv2, tv1), where tv2 only passes cofactorless) is the reverse of the usual (tv1, tv2), which may induce a confusion.

Reply:
Yeh, the logic here was to have vectors 0-3 to satisfy both cofactored and cofactorless.